### PR TITLE
Display Classifications filter as a tree

### PR DESCRIPTION
**Pull request by cesarvarela**
https://pr-9--cesarvarela-staging.netlify.app/apps/discover/?classifications=CSETv0%3ALocation%3APalo%20Alto%2C%20CA&is_incident_report=true

### CODE REVIEW

**Comment by kepae** [comment]
Want to sync and try this with https://github.com/responsible-ai-collaborative/aiid/pull/2508?

**Comment by kepae** [comment]
Empty facets in this latest preview?

image


I'm tired at the moment and maybe missing something... but worth checking behavior on env variables and if the algolia index for staging is being passed, etc.

**Comment by cesarvarela** [comment]
@kepae ready for another review

latest passing build: https://github.com/aiidtest/aiid/actions/runs/7455349588

File: site/gatsby-site/src/utils/AlgoliaUpdater.js [hunk 1]
```diff
@@ -45,7 +47,9 @@ const includedCSETAttributes = [
 ];
 
 const getClassificationArray = ({ classification, taxonomy }) => {
```
Comment (kepae) [inline site/gatsby-site/src/utils/AlgoliaUpdater.js:49]: please consider renaming, especially now that an object is being returned. Even better would be providing an comment example with the return type, considering the readability of `list` and the return object.

File: site/gatsby-site/src/utils/AlgoliaUpdater.js [hunk 2]
```diff
@@ -112,8 +120,16 @@ const reportToEntry = ({ incident = null, report }) => {
     featured: featuredValue,
     flag: report.flag,
     is_incident_report: report.is_incident_report,
+    namespaces: classifications.map((c) => {
+      return Object.keys(c.tree)[0];
```
Comment (kepae) [inline site/gatsby-site/src/utils/AlgoliaUpdater.js:124]: is it fair to assume the keys of the `classification.tree` are ordered here?

File: site/gatsby-site/src/components/discover/Filters.js [hunk 3]
```diff
@@ -2,10 +2,26 @@ import React from 'react';
...
+import { graphql, useStaticQuery } from 'gatsby';
 
 function Filters() {
   const { isCollapsed } = useMenuContext();
 
+  const {
+    taxa: { nodes: taxa },
+  } = useStaticQuery(graphql`
+    query FiltersTaxaQuery {
+      taxa: allMongodbAiidprodTaxa(filter: { namespace: { in: ["CSETv1", "CSETv0", "GMF"] } }) {
```
Comment (kepae) [inline site/gatsby-site/src/components/discover/Filters.js:14]: for later: we should identify all the places where taxa are hard-coded and move it into the config (as started in this PR).

File: site/gatsby-site/src/components/discover/filterTypes/Classifications.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+          })}
+        </div>
+      ) : (
+        <div className="mt-3">No results</div>
+      )}
+    </div>
+  );
+}
+
+export const touchedCount = ({ searchState }) => {
```
Comment (kepae) [inline site/gatsby-site/src/components/discover/filterTypes/Classifications.js:257]: unused? (or I can't find it in code search, my bad)

File: site/gatsby-site/src/components/discover/parseURL.js [hunk 5]
```diff
@@ -14,21 +14,33 @@ const convertStringToRefinement = (obj) => {
...
+          for (const facet of facets) {
+            const [namespace, attribute, ...value] = facet.split(':');
+
+            const refinementKey = `${namespace}.${attribute}`;
+
+            if (!result[refinementKey]) {
+              result[refinementKey] = [];
+            }
+
+            result[`${namespace}.${attribute}`].push(...value);
```
Comment (kepae) [inline site/gatsby-site/src/components/discover/parseURL.js:34]: *`result[refinementKey]`?

**Review by kepae — COMMENTED** [review]
Minor comments then good to go!

File: site/gatsby-site/src/utils/AlgoliaUpdater.js (hunk 2 above)
Comment (cesarvarela) [reply site/gatsby-site/src/utils/AlgoliaUpdater.js:124]: It is more that the order doesn't matter.

File: site/gatsby-site/src/components/discover/filterTypes/Classifications.js (hunk 4 above)
Comment (cesarvarela) [reply site/gatsby-site/src/components/discover/filterTypes/Classifications.js:257]: Those are used here:
https://github.com/responsible-ai-collaborative/aiid/blob/ed22c006d03a36b3752fedc17b6ce48f7bb43733/site/gatsby-site/src/components/discover/Filter.js#L144

File: site/gatsby-site/src/components/discover/Filters.js (hunk 3 above)
Comment (cesarvarela) [reply site/gatsby-site/src/components/discover/Filters.js:14]: Sadly, that's a limitation of Gatsby's `useStaticQuery.` [You can't do dynamic queries](https://www.gatsbyjs.com/docs/how-to/querying-data/use-static-query/#differences-between-page-queries--static-queries).

We should refactor that one eventually to use a page query or dynamically pull from the graphql endpoint.

**Comment by cesarvarela** [comment]
@kepae ready for another review, the failing test won't pass until this is merged to staging and the algolia index is updated

File: site/gatsby-site/src/components/discover/filterTypes/Classifications.js (hunk 4 above)
Comment (kepae) [reply site/gatsby-site/src/components/discover/filterTypes/Classifications.js:257]: Thanks, I figured I was likely missing it.

**Review by kepae — APPROVED** [review]

**Pull request merged**